src/wallet/wallet.cpp

```diff
@@ -2404,18 +2408,15 @@ bool CWallet::FundTransaction(CMutableTransaction& tx, CAmount& nFeeRet, bool ov
     if (!CreateTransaction(vecSend, wtx, vpChangeKey, nFeeRet, nChangePosInOut, strFailReason, &coinControl, false))
         return false;
 
-    // Append fee output if any
-    // This assumes fee is appended to end, and only one fee added by the CreateTransaction call
-    if (wtx.vout.back().IsFee()) {
-        tx.vout.push_back(wtx.vout.back());
-    }
-
-    if (nChangePosInOut != -1) {
-        tx.vout.insert(tx.vout.begin() + nChangePosInOut, wtx.vout[nChangePosInOut]);
-
-        // Insert change witness
-        tx.wit.vtxoutwit.resize(tx.vout.size()-1);
-        tx.wit.vtxoutwit.insert(tx.wit.vtxoutwit.begin() + nChangePosInOut,  wtx.wit.vtxoutwit[nChangePosInOut]);
+    // Wipe outputs and output witness and re-add one by one
+    wtx.wit.vtxoutwit.resize(wtx.vout.size());
```

Comment:
we're iterating over the size of the vout, and I just want to make sure a witness actually exists, blank or not for each.